src/assets/rewardrequestdb.cpp

```diff
@@ -0,0 +1,282 @@
+    }
+
+    //  Overwrite the entry in the database
+    bool succeeded = Write(std::make_pair(SCHEDULEDREWARD_FLAG, p_newReward.heightForPayout), reqDbEntry);
+
+    LogPrintf("%s : Scheduling reward %s %s!\n",
+        __func__,
+        p_newReward.rewardID.c_str(),
+        succeeded ? "succeeded" : "failed");
+
+    return succeeded;
+}
+
+bool CRewardRequestDB::RetrieveRewardWithID(
+    const std::string & p_rewardID, CRewardRequest & p_reward)
+{
+    LogPrintf("%s : Looking for scheduled reward with ID '%s'\n",
+        __func__, p_rewardID.c_str());
+
+    std::unique_ptr<CDBIterator> pcursor(NewIterator());
+
+    pcursor->SeekToFirst();
+
+    // Load all pending rewards
+    while (pcursor->Valid()) {
+        boost::this_thread::interruption_point();
+        std::pair<char, int> key;
+
+        if (pcursor->GetKey(key) && key.first == SCHEDULEDREWARD_FLAG) {
+            std::set<CRewardRequest> dbEntrySet;
+
+            if (pcursor->GetValue(dbEntrySet)) {
+                LogPrintf("%s: Current set has %d entries\n",
+                    __func__, dbEntrySet.size());
+                for (auto const & entry : dbEntrySet) {
+                    //  Only retrieve the specified reward
+                    if (p_rewardID == entry.rewardID) {
+                        LogPrintf("%s : Found reward: rewardID='%s', wallet='%s', height=%d, amt=%lld, srcAsset='%s', tgtAsset='%s', exceptions='%s'\n",
+                            __func__,
+                            entry.rewardID.c_str(),
+                            entry.walletName.c_str(), entry.heightForPayout, static_cast<long long>(entry.totalPayoutAmt),
+                            entry.payoutSrc.c_str(), entry.tgtAssetName.c_str(),
+                            entry.exceptionAddresses.c_str());
+
+                        p_reward = entry;
+
+                        return true;
+                    }
+                }
+            } else {
+                LogPrintf("%s: Failed to read reward\n", __func__);
+            }
+        }
+
+        pcursor->Next();
+    }
+
+    return false;
+}
+
+bool CRewardRequestDB::RemoveReward(const std::string & p_rewardID)
+{
+    LogPrintf("%s : Attempting to remove reward: rewardID='%s'\n",
+        __func__,
+        p_rewardID.c_str());
+
+    CRewardRequest rewardToRemove;
+    if (!RetrieveRewardWithID(p_rewardID, rewardToRemove)) {
+        LogPrintf("%s : Failed to find reward: rewardID='%s'!\n",
+            __func__,
+            p_rewardID.c_str());
+
+        return false;
+    }
+    //  Retrieve height-based entry which contains list of rewards to pay at that height
+    CRewardRequestDBEntry reqDbEntry;
+
+    //  If this fails, bail, since we can't be sure we won't blow away valid entries
+    if (!Read(std::make_pair(SCHEDULEDREWARD_FLAG, rewardToRemove.heightForPayout), reqDbEntry)) {
+        LogPrintf("%s : Failed to read scheduled rewards at specified height!\n", __func__);
```

Comment:
This is another issue we run into when using sets that contain multiple items for the height as the value in the database

Reply:
Resolved by changing key to a string concatenation of block height and asset name.